### ISSUE TITLE
URL missing application virtual folder in IIS

### ISSUE THREAD

**Issue by schardag** [LazZiya/TagHelpers#26]
When hosting an application in IIS the application virtual folder is missing from the url created in the paging tag helper. 

example:
IIS hosted application url
https://`<hosting url>`/`<application virtual folder>`/Admin/Applications

When you click a pager link the url is with the application virtual folder missing:
https://`<hosting url>`/Admin/Applications?p=3&s=25

**Expected behavior**
The applications virtual folder is included in the link which will create a fully functional url.

https://`<hosting url>`/`<application virtual folder>`/Admin/Applications?p=3&s=25

**Comment by LazZiya** [comment]
Thanks @schardag for reporting this issue, I will check and provide a fix soon...

**Comment by schardag** [comment]
I have found out that if I use fix-url-path="false" it fixes the issue. Is this the proper solution?

**Comment by LazZiya** [comment]
Good to know that it fixed the issue, you saved me some research time :)

The main purpose of `fix-url-path` is to force adding the page path to the paging links, espcially if the page is located inside area and/or under sub folders, and it is `true` by default:

https://github.com/LazZiya/TagHelpers/blob/c424b3a66c2c41dd2dc6147adcd1ad62333f3d3a/LazZiya.TagHelpers/PagingTagHelper.cs#L83-L89

https://github.com/LazZiya/TagHelpers/blob/c424b3a66c2c41dd2dc6147adcd1ad62333f3d3a/LazZiya.TagHelpers/PagingTagHelper.cs#L865-L870

So if everything is workng fine with setting `fix-url-path=false` then it is the proper solution.

Best,
Ziya

**Comment by Malixoxo** [comment]
> Good to know that it fixed the issue, you saved me some research time :)
> 
> The main purpose of `fix-url-path` is to force adding the page path to the paging links, espcially if the page is located inside area and/or under sub folders, and it is `true` by default:
> 
> https://github.com/LazZiya/TagHelpers/blob/c424b3a66c2c41dd2dc6147adcd1ad62333f3d3a/LazZiya.TagHelpers/PagingTagHelper.cs#L83-L89
> 
> https://github.com/LazZiya/TagHelpers/blob/c424b3a66c2c41dd2dc6147adcd1ad62333f3d3a/LazZiya.TagHelpers/PagingTagHelper.cs#L865-L870
> 
> So if everything is workng fine with setting `fix-url-path=false` then it is the proper solution.
> 
> Best, Ziya

Hi @LazZiya,

first of all thanks for the Helper. Its really helpful!
...but I have the same problem and I dont know how to fix that?

When I click on the paging links or hover, the application virtual folder is missing and causing a 404 error.

**Comment by LazZiya** [comment]
Hi @Malixoxo , 

Glad to know it is helping you :)

Did you tried setting the parameter for fixing url? 

`fix-url-path=false`

**Comment by Malixoxo** [comment]
Hi @LazZiya 

I must have missed that point in the documentation sorry.
I did that and it works thanks!

Sorry for brothering you with that.

**Comment by LazZiya** [comment]
no problem, you are welcome @Malixoxo :)

so, I'm closing the issue since it is confirmed twice the problem already has a solution 👍🏽

**Comment by Pucc1o** [comment]
I tried both configurations for fix-url-path and it is not working for me under IIS.

**Comment by LazZiya** [comment]
@Pucc1o can you share more details? configuration?

**Comment by Pucc1o** [comment]
IIS hosted application 
https://`<hosting url>`/`<application virtual folder>`/Admin/SamplePage

When you click a pager link or the page size dropdown the url is with the application virtual folder missing:
https://`<hosting url>`/Admin/SamplePage?p=3&s=25

adding the `fix-url-path=false` the subfolder is missing
https://`<hosting url>`/`<application virtual folder>`/?p=3&s=25

**Comment by Pucc1o** [comment]
I think I have found the issue. 
My solution implements Blazor Server Side and Razor Pages, which requires to define `<base href="/" />` to specify the root of the Blazor app.
Removing the  `<base href="/" />` makes it work with the paging tag helper but breaks the Blazor app.
I looked at possible alternative solutions but there isn't one and Microsoft position is 
> `<base href="<<something>>"` is the standard way to make this work

Ideas?
